SingularityUI/app/components/common/modalButtons/RedeployButton.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { Glyphicon } from 'react-bootstrap';
+import OverlayTrigger from 'react-bootstrap/lib/OverlayTrigger';
+import ToolTip from 'react-bootstrap/lib/Tooltip';
+
+import { getClickComponent } from '../modal/ModalWrapper';
+import { FetchDeployForRequest } from '../../../actions/api/history';
+
+import RedeployModal from './RedeployModal';
+import Utils from '../../../utils';
+
+const redeployTooltip = (
+  <ToolTip id="redeploy">
+    Redeploy
+  </ToolTip>
+);
+
+class RedeployButton extends Component {
```

Comment:
Yeah that's actually related to a problem I've run into, where I want to use the buttons in a dropdown but clicking anything in the modal closes it because the modal is a child of the dropdown.